app/work_template_executor.go

```diff
@@ -0,0 +1,273 @@
+	}
+	resp, appErr = e.app.doPluginRequest(c, http.MethodPost, "/plugins/playbooks/api/v0/runs", nil, data)
+	if appErr != nil {
+		return "", fmt.Errorf("unable to create playbook run: %w", appErr)
+	}
+	defer resp.Body.Close()
+	pbrResp := playbookRunCreateResponse{}
+	err = json.NewDecoder(resp.Body).Decode(&pbrResp)
+	if err != nil {
+		return "", fmt.Errorf("unable to decode playbook run create response: %w", err)
+	}
+
+	// using pbrResp.ChannelID, update the channel to add metadata
+	dbChannel, err := e.app.Srv().Store().Channel().Get(pbrResp.ChannelID, false)
+	if err != nil {
+		return "", fmt.Errorf("unable to find channel: %w", err)
+	}
+	if dbChannel == nil {
+		return "", fmt.Errorf("channel not found")
+	}
+	dbChannel.AddProp(model.WorkTemplateIDChannelProp, wtcr.WorkTemplate.ID)
+	_, err = e.app.Srv().Store().Channel().Update(dbChannel)
+	if err != nil {
+		e.app.Srv().Log().Error("Failed to update playbook channel metadata", mlog.Err(err))
+	}
+
+	return pbrResp.ChannelID, nil
+}
+
+func (e *appWorkTemplateExecutor) CreateChannel(
+	c *request.Context,
+	wtcr *worktemplates.ExecutionRequest,
+	cChannel *model.WorkTemplateChannel,
+) (string, error) {
+	channelID := ""
+	channelDisplayName := cChannel.Name
+	if wtcr.Name != "" {
+		channelDisplayName = fmt.Sprintf("%s: %s", wtcr.Name, cChannel.Name)
+	}
+
+	var channelCreationAppErr *model.AppError = &model.AppError{}
+	cleanChannelName := cleanChannelName(channelDisplayName)
+	channelName := cleanChannelName
+	if len(channelName) > model.ChannelNameMaxLength {
+		channelName = channelName[:model.ChannelNameMaxLength]
+	}
+	for channelCreationAppErr != nil {
+		// create channel
+		newChan, channelCreationAppErr := e.app.CreateChannelWithUser(c, &model.Channel{
+			TeamId:      wtcr.TeamID,
+			Name:        channelName,
+			DisplayName: channelDisplayName,
+			Type:        model.ChannelTypeOpen,
+			Purpose:     cChannel.Purpose,
+			Props: map[string]any{
+				model.WorkTemplateIDChannelProp: wtcr.WorkTemplate.ID,
+			},
+		}, c.Session().UserId)
+		channelID = newChan.Id
+		if channelCreationAppErr != nil {
+			if channelCreationAppErr.Id == store.ChannelExistsError {
+				// compute a new unique name
+				suffix := fmt.Sprintf("-%s", model.NewId()[0:4])
+				channelName = cleanChannelName
+				if len(cleanChannelName)+len(suffix) > model.ChannelNameMaxLength {
+					channelName = cleanChannelName[:model.ChannelNameMaxLength-len(suffix)]
+				}
+				channelName = channelName + suffix
+				continue
+			}
+
+			return "", fmt.Errorf("error while creating channel: %w", channelCreationAppErr)
+		}
+	}
+
+	return channelID, nil
+}
+
+func (e *appWorkTemplateExecutor) CreateBoard(
+	c *request.Context,
+	wtcr *worktemplates.ExecutionRequest,
+	cBoard *model.WorkTemplateBoard,
+	linkToChannelID string,
+) (string, error) {
+	boardService := e.app.Srv().services[product.BoardsKey].(product.BoardsService)
+	templates, err := boardService.GetTemplates("0", c.Session().UserId)
```

Comment:
Yeah focalboards uses team ID `"0"` to register Mattermost templates.

Reply:
nit: to avoid this magic number/string place that in a const with a meaningful name 👍

Reply:
Yeah I agree but this where the border between product/server becomes blurry, I'm not sure where this constant belongs to 😅 Should this be provided by focalboard? It might actually exist but I don't know if we're allowed to invoke focalboard packages from here (if we don't build in product mode for example)